src/main/java/org/jboss/netty/handler/ssl/SslHandler.java

```diff
@@ -1284,6 +1278,20 @@ private ChannelBuffer unwrap(
 
                     if (result.getStatus() == Status.BUFFER_UNDERFLOW ||
                         result.bytesConsumed() == 0 && result.bytesProduced() == 0) {
+                        if (nioInNetBuf.hasRemaining() && !engine.isInboundDone()) {
+                            // We expect SSLEngine to consume all the bytes we feed it, but
+                            // empirical evidence indicates that we sometimes end up with leftovers
+                            // Log when this happens to get a better understanding of this corner
+                            // case.
+                            // See https://github.com/netty/netty/pull/3584
+                            logger.warn("unexpected leftover data after SSLEngine.unwrap: "
```

Comment:
* unexpected -> Unexpected
* SSLEngine.unwrap -> SSLEngine.unwrap()
* The last space in this line is not necessary because the next line starts with a space.